src/lualib/Error.ts

```diff
@@ -0,0 +1,56 @@
+interface ErrorType<T> extends Function {
+    name: string;
+    new (...args: any[]): T;
+}
+
+function __TS__GetErrorStack(constructor: Function): string {
+    let level = 1;
+    while (true) {
+        const info = debug.getinfo(level, "f");
+        level += 1;
+        if (!info) {
+            // constructor is not in call stack
+            level = 1;
+            break;
+        } else if (info.func === constructor) {
+            break;
+        }
+    }
+    return debug.traceback(undefined, level);
+}
+
+function __TS__GetErrorString(this: void, error: Error): string {
+    return error.message !== "" ? `${error.name}: ${error.message}` : error.name;
+}
+
+function __TS__InitErrorClass<T>(Type: ErrorType<T>, name?: string): any {
```

Comment:
Type of `Type` there doesn't make any code more sound since generic isn't used in return type, so I don't see any advantages plain `any` there. Actually it could be typed correctly:

```ts
type AddCallSignature<T> = T extends new (...args: infer TArgs) => infer TReturn
    ? T & ((...args: TArgs) => TReturn)
    : never;

function __TS__InitErrorClass<T>(Type: T, name?: string): AddCallSignature<T> {
    if (name) {
        Type.name = name;
    }

    return setmetatable(Type, {
        __call: (_self: any, message: string) => new Type(message),
    });
}

__TS__InitErrorClass(class Error {}); // => typeof Error & (() => Error)
```

But I'm not really sure if adding a quite non trivial type there worth it.

Reply:
I'm not sure that would work. How would TS know that a call method has been added by `setmetatable` without an explicit cast?

Reply:
Oh, I have assumed `setmetatable` returns `any`. Yeah, now it wouldn't work without a cast on returned value. It might make sense to add it to `setmetatable` signature instead (in `lua-types` if we'll switch to it).